pkg/transform/mapper.go

```diff
@@ -0,0 +1,156 @@
+/**
+ * Copyright (c) 2020-present Snowplow Analytics Ltd.
+ * All rights reserved.
+ *
+ * This software is made available by Snowplow Analytics, Ltd.,
+ * under the terms of the Snowplow Limited Use License Agreement, Version 1.0
+ * located at https://docs.snowplow.io/limited-use-license-1.0
+ * BY INSTALLING, DOWNLOADING, ACCESSING, USING OR DISTRIBUTING ANY PORTION
+ * OF THE SOFTWARE, YOU AGREE TO THE TERMS OF SUCH LICENSE AGREEMENT.
+ */
+
+package transform
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"time"
+
+	"github.com/itchyny/gojq"
+
+	"github.com/snowplow/snowbridge/config"
+	"github.com/snowplow/snowbridge/pkg/models"
+)
+
+// JQMapperConfig represents the configuration for the JQ transformation
+type JQMapperConfig struct {
+	JQCommand  string `hcl:"jq_command"`
+	RunTimeout int    `hcl:"timeout_sec,optional"`
```

Comment:
2 small nits on this:

- 1s is actually quite a long minimum possible timeout for this kind of feature. I'd expect it to be very fast - perhaps it should be in ms?
- Could we have the unit in the setting name? Just for QOL